quinn-proto/src/congestion.rs

```diff
@@ -77,7 +78,7 @@ pub trait Controller: Send {
 /// Constructs controllers on demand
 pub trait ControllerFactory {
     /// Construct a fresh `Controller`
-    fn build(&self, now: Instant, current_mtu: u16) -> Box<dyn Controller>;
+    fn build(self: Arc<Self>, now: Instant, current_mtu: u16) -> Box<dyn Controller>;
```

Comment:
`&Arc<Self>` might be even nicer here, in case a future impl doesn't want to capture a reference to `self`, but that's not (yet?) object safe and all our actual impls want this pattern anyway.